src/main/scala/format/pgn/Parser.scala

```diff
@@ -91,7 +91,7 @@ object Parser extends scalaz.syntax.ToTraverseOps {
       }
     }
 
-    val moveRegex = """0\-0\-0|0\-0|[PQKRBNOoa-h@][QKRBNa-h1-8xOo\-=\+\#\@]{1,6}[\?!□]{0,2}""".r
+    val moveRegex = """(?:0\-0\-0[+#]?|0\-0[+#]?|[PQKRBNOoa-h@][QKRBNa-h1-8xOo\-=\+\#\@]{1,6})[\?!□]{0,2}""".r
```

Comment:
Perhaps this thrice-repeated `[+#]?` could be moved outside the parentheses.

Reply:
Several SAN parsing regexp solutions exists out there, like
https://github.com/pychess/pychess/blob/master/lib/pychess/Savers/pgn.py#L55
or
https://github.com/niklasf/python-chess/blob/master/chess/pgn.py#L87
but my intention was to make as little change as possible to fix this issue only.